src/Aspire.Hosting/Publishing/ManifestPublishingContext.cs

```diff
@@ -101,9 +101,9 @@ public async Task WriteContainerAsync(ContainerResource container)
     public void WriteConnectionString(IResource resource)
     {
         if (resource is IResourceWithConnectionString resourceWithConnectionString &&
-            resourceWithConnectionString.ConnectionStringExpression is string connectionString)
+            resourceWithConnectionString.ConnectionStringExpression is { } connectionString)
```

Comment:
This feels like we shouldn't need a null check. The property is defined as `public ReferenceExpression ConnectionStringExpression { get; }`, so it should never be null.  (and I also really dislike this syntax 😢).